source/MRMesh/MRFeatureObject.cpp

```diff
@@ -54,6 +55,30 @@ void FeatureObject::deserializeFields_( const Json::Value& root )
 
     if ( const auto& detailsOnNameTagJson = root["DetailsOnNameTag"]; detailsOnNameTagJson.isUInt() )
         detailsOnNameTag_ = ViewportMask( detailsOnNameTagJson.asUInt() );
+
+    // only default xf value serialyze now.
+    decomposeMatrix3( xf().A, r_.get(), s_.get() );
+}
+
+std::optional<Vector3f> FeatureObject::getNormal( const Vector3f& point ) const
+{
+    return projectPoint( point ).normal;
+}
+
+void FeatureObject::setXf( const AffineXf3f& xf, ViewportId id )
+{
+
+    decomposeMatrix3( xf.A, r_[id], s_[id] );
+    VisualObject::setXf( xf, id );
```

Comment:
better to compare same if old xf is equivalent to new one and fast return

Reply:
agree